dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/handlers/MasterThesisPdfAttestationGeneratorHandler.java

```diff
@@ -0,0 +1,136 @@
+            InputStream xmlDataInputStream = new ByteArrayInputStream(renderedXml.getBytes());
+            StreamSource xmlDataSource = new StreamSource(xmlDataInputStream);
+
+            //2. Instantiate FOP Factory 
+            FopFactory fopFactory = FopFactory.newInstance(new File(".").toURI());
+            FOUserAgent foUserAgent = fopFactory.newFOUserAgent();
+
+            //3. Use transformer to create the final PDF
+            try {
+                // Construct FOP 
+                Fop fop = fopFactory.newFop(MimeConstants.MIME_PDF, foUserAgent, out);
+                TransformerFactory factory = TransformerFactory.newInstance();
+                Transformer transformer = factory.newTransformer(new StreamSource(templateFile));
+
+                Result res = new SAXResult(fop.getDefaultHandler());
+
+                transformer.transform(xmlDataSource, res);
+            } finally {}
+        } catch (Exception e) {
+            System.out.println(e);
+            throw new PDFGenerationException(e.getMessage());
+        }
+    };
+
+    /** 
+    * Generates a XML containing data to feed the template file.
+    *
+    * @param uuid: UUID of the targeted DSpace object.
+    * @param DSpaceContext: The current DSpace context, used to recover the DSpace object from UUID.
+    */
+    private String generatePDFMasterThesisAttestationFromObjectId(UUID uuid, Context DSpaceContext) throws SQLException {
+        // 1. Retrieve DSpace item's metadata
+        Item dspaceItem = itemService.find(DSpaceContext, uuid);
+        List<MetadataValue> metadataValues = dspaceItem.getMetadata();
+        MasterThesisPDFAttestationModel pdfModel = new MasterThesisPDFAttestationModel();
+
+        HashMap<String, List<String>> map = metadataUtils.getValuesHashMap(metadataValues);
+
+        // 2. Add information to the model
+        pdfModel.title = map.get("dc_title").get(0);
+
+        // Add authors to model
+        for (String authorName: map.get("dc_contributor_author")) {
+            pdfModel.addAuthor(authorName);
+        }
+        // Add advisors to model
+        for (String advisorName: map.get("dc_contributor_advisor")) {
+            pdfModel.addAdvisor(advisorName);
+        }
+        // Add programs to model 
+        for (String programName: map.get("masterthesis_degree_code")) {
+            pdfModel.addProgram(programName);
+        }
+
+        pdfModel.submitter = dspaceItem.getSubmitter().getFullName();
+        // TODO: Manage Handle recovery
+        pdfModel.handle = "http://handle.net/";
+        
+        // Add files to model
+        // TODO: Recover permissions for each files ?? => For now, HARD CODED.
+        for (Bitstream bitstream: metadataUtils.extractItemFiles(dspaceItem)) {
+            pdfModel.addFile(bitstream.getName(), "Open access");
+        }
+
+        pdfModel.abstractText = map.get("dc_description_abstract").get(0);
+        pdfModel.imagePath = this.source + "/assets/images/dial_mem.png"; 
```

Comment:
COSMETIC : maybe assets for PDFGeneration (customCss, images, ...) could be placed into a dedicated directory instead. I don't know the best practice.... maybe it's overkill

Reply:
For now I think we can keep it in there but in the future who knows o_O ?